### PR TITLE
Issue #2498 Removed static lock from buffermanager impl

### PR DESCRIPTION
**Pull request by Arjun-Sharma1**
Also moved EMQ to its own class with separate lock
#2498

### CODE REVIEW

File: dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/EMQController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package com.ibm.ws.collector.manager.buffer;
+
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import com.ibm.wsspi.collector.manager.BufferManager;
+
+public class EMQController {
+
+    protected final static ReentrantReadWriteLock RERWLOCK = new ReentrantReadWriteLock(true);
```
Comment (raymondlam) [inline dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/EMQController.java]: I don't think u need a read-write lock for this. You can just synchronize on the list

File: dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/BufferManagerImpl.java [hunk 2]
```diff
@@ -22,27 +22,25 @@
...
-    private static final ReentrantReadWriteLock RERWLOCK = new ReentrantReadWriteLock(true);
+    private final ReentrantReadWriteLock RERWLOCK = new ReentrantReadWriteLock(true);
     private Set<SynchronousHandler> synchronousHandlerSet = new HashSet<SynchronousHandler>();
     private final int capacity;
 
     private final String sourceId;
     /* Map to keep track of the next event for a handler */
     private final ConcurrentHashMap<String, HandlerStats> handlerEventMap = new ConcurrentHashMap<String, HandlerStats>();
-    private static List<BufferManager> bufferManagerList= new ArrayList<BufferManager>();
+    static List<BufferManager> bufferManagerList= Collections.synchronizedList(new ArrayList<BufferManager>());
```
Comment (raymondlam) [inline dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/BufferManagerImpl.java]: We should have a quick discussion on whether the list should be in BufferManagerImpl vs EMQController. EMQController makes it cleaner to handle concurrency.

File: dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/BufferManagerEMQHelper.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                    public void run() {
+                        BufferManagerEMQHelper.removeEMQTrigger();
+                    }
+                },
+                BufferManagerEMQHelper.EMQ_TIMER);
+    }
+    
+    public synchronized static void removeEMQTrigger(){
+            BufferManagerEMQHelper.EMQRemovedFlag=true;
+            for(BufferManager i: BufferManagerEMQHelper.bufferManagerList) {
```
Comment (raymondlam) [inline dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/BufferManagerEMQHelper.java]: You need to have `synchronize(bufferManagerList)` for this to make this thread safe

Comment (Arjun-Sharma1) [reply dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/BufferManagerEMQHelper.java]: Created an add method in BufferManagerEMQHelper which is synchronized and adds to the bufferManagerList.

**Comment by Arjun-Sharma1** [comment]
#build

**Comment by Arjun-Sharma1** [comment]
#build

**Pull request merged**